### PR TITLE
Update prometheus-common and the consul HTTP client

### PR DESCRIPTION
**Pull request by austince**
Updates the consul_sd to the prometheus-common HTTP client, using the new `config.WithIdleConnTimeout` func. 

Not sure if you actually want this in @roidelapluie, but as promised if you do :) 

Will update the prometheus-common version as soon as a new version is tagged.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -47,9 +47,9 @@ require (
 	github.com/opentracing/opentracing-go v1.2.0
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/alertmanager v0.21.0
-	github.com/prometheus/client_golang v1.10.0
+	github.com/prometheus/client_golang v1.11.0
 	github.com/prometheus/client_model v0.2.0
-	github.com/prometheus/common v0.23.0
+	github.com/prometheus/common v0.28.1-0.20210609145511-8281fb2779ab
```
Comment (roidelapluie) [inline go.mod:52]: I have tagged 0.29

Comment (austince) [reply go.mod:52]: Thanks!

File: discovery/consul/consul.go [hunk 2]
```diff
@@ -184,22 +182,16 @@ func NewDiscovery(conf *SDConfig, logger log.Logger) (*Discovery, error) {
 		logger = log.NewNopLogger()
 	}
 
-	tls, err := config.NewTLSConfig(&conf.TLSConfig)
+	httpConfig := config.HTTPClientConfig{
+		TLSConfig: conf.TLSConfig,
```
Comment (roidelapluie) [inline discovery/consul/consul.go:186]: As a temporary workaround, can you add `FollowRedirects: true,`, so we do not break users?

Comment (austince) [reply discovery/consul/consul.go:186]: Yes, I wondered about that, misread the `client.CheckRedirect` configuration – thanks for the catch.

**Review by roidelapluie — APPROVED** [review]

**Comment by austince** [comment]
Resolved conflicts

**Pull request merged**